engine/ui/ui_router.go

```diff
@@ -29,41 +29,80 @@ func (s *Service) initRouter(ctx context.Context) {
 	r.Handle(s.Cfg.DeployURL+"/mon/metrics", nil, r.GET(service.GetPrometheustMetricsHandler(s)))
 	r.Handle(s.Cfg.DeployURL+"/mon/metrics/all", nil, r.GET(service.GetMetricsHandler))
 
-	// proxypass
-	if s.Cfg.API.HTTP.URL != "" {
-		r.Mux.PathPrefix(s.Cfg.DeployURL + "/cdsapi").Handler(s.getReverseProxy(s.Cfg.DeployURL+"/cdsapi", s.Cfg.API.HTTP.URL))
-	}
-	if s.Cfg.HooksURL != "" {
-		r.Mux.PathPrefix(s.Cfg.DeployURL + "/cdshooks").Handler(s.getReverseProxy(s.Cfg.DeployURL+"/cdshooks", s.Cfg.HooksURL))
-	}
-	if s.Cfg.CDNURL != "" {
-		r.Mux.PathPrefix(s.Cfg.DeployURL + "/cdscdn").Handler(s.getReverseProxy(s.Cfg.DeployURL+"/cdscdn", s.Cfg.CDNURL))
+	// proxypass if enabled
+	if s.Cfg.EnableServiceProxy {
+		if s.Cfg.API.HTTP.URL != "" {
+			apiPath := s.Cfg.DeployURL + "/cdsapi"
+			r.Mux.PathPrefix(apiPath).Handler(s.getReverseProxy(ctx, apiPath, s.Cfg.API.HTTP.URL))
+		}
+		if s.Cfg.HooksURL != "" {
+			hooksPath := s.Cfg.DeployURL + "/cdshooks"
+			r.Mux.PathPrefix(hooksPath).Handler(s.getReverseProxy(ctx, hooksPath, s.Cfg.HooksURL))
+		}
+		if s.Cfg.CDNURL != "" {
+			cdnPath := s.Cfg.DeployURL + "/cdscdn"
+			r.Mux.PathPrefix(cdnPath).Handler(s.getReverseProxy(ctx, cdnPath, s.Cfg.CDNURL))
+		}
 	}
 
 	// serve static UI files
 	r.Mux.PathPrefix("/docs").Handler(s.uiServe(http.Dir(s.DocsDir), s.DocsDir))
 	r.Mux.PathPrefix("/").Handler(s.uiServe(http.Dir(s.HTMLDir), s.HTMLDir))
 }
 
-func (s *Service) getReverseProxy(path, urlRemote string) *httputil.ReverseProxy {
-	origin, _ := url.Parse(urlRemote)
+func (s *Service) getReverseProxy(ctx context.Context, path, urlRemote string) http.Handler {
+	filter := func(req *http.Request) bool {
+		reqPath := strings.TrimPrefix(req.URL.Path, path)
+
+		// on api proxypass, deny request on /mon/metrics
+		if strings.HasSuffix(path, "api") && strings.HasPrefix(reqPath, "/mon/metrics") {
+			return false
+		}
+
+		// on hooks proxypass, allow only request on /webhook/ path
+		if strings.HasSuffix(path, "hooks") && !strings.HasPrefix(reqPath, "/webhook/") {
+			return false
+		}
+
+		// on cdn proxypass, allow only request on /item
+		if strings.HasSuffix(path, "cdn") && !strings.HasPrefix(reqPath, "/item") {
+			return false
+		}
 
+		return true
+	}
+
+	origin, _ := url.Parse(urlRemote)
 	director := func(req *http.Request) {
 		reqPath := strings.TrimPrefix(req.URL.Path, path)
-		// on proxypass /cdshooks, allow only request on /webhook/ path
-		if strings.HasSuffix(path, "/cdshooks") && !strings.HasPrefix(reqPath, "/webhook/") {
-			// return 502 bad gateway
-			req = &http.Request{} // nolint
-		} else {
-			req.Header.Add("X-Forwarded-Host", req.Host)
-			req.Header.Add("X-Origin-Host", origin.Host)
-			req.URL.Scheme = origin.Scheme
-			req.URL.Host = origin.Host
-			req.URL.Path = origin.Path + reqPath
-			req.Host = origin.Host
-		}
+		req.Header.Add("X-Forwarded-Host", req.Host)
```

Comment:
may be use the `HeaderXForwardedFor` value from the configuration

Reply:
Will do this